src/apps/replication/meta_server/meta_service.cpp

```diff
@@ -118,6 +112,12 @@ error_code meta_service::start()
         return err;
     }
 
+    _balancer = new load_balancer(_state);
+    // make sure the delay is larger than fd.grace to ensure
+    // all machines are in the correct state (assuming connected initially)
+    tasking::enqueue(LPC_LBM_START, this, &meta_service::on_load_balance_start, 0,
+        _opts.fd_grace_seconds * 1000);
+
     // register rpc handlers
     register_rpc_handler(
         RPC_CM_QUERY_NODE_PARTITIONS,
```

Comment:
register_rpc_handler must be before acqutie_leader_lock above so that the service can tell the others who is the current leader. this was fixed before, but seems rollback-ed during merge operations.